From 6599365a2a17d2c26777d61234d55ab01f3b88e5 Mon Sep 17 00:00:00 2001
From: P Praneesh <quic_ppranees@quicinc.com>
Date: Sat, 21 Sep 2024 12:18:56 +0530
Subject: [PATCH] wifi: ath12k: Fix double free while setting packet len

When pskb_expand_head fails while setting packet length, skb is getting
freed in ath12k_dp_pkt_set_pktlen as well as caller which leads to
double free. Fix it by removing the skb free from
ath12k_dp_pkt_set_pktlen.

Fixes: (Ib4b65722c) ath12k: add monitor buffer address TLV parsing

Patch-dependency: 640-001-ath12k-Enable-monitor-interface-support.patch

Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp_mon.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/dp_mon.c b/drivers/net/wireless/ath/ath12k/dp_mon.c
index 854f0fc..2d49235 100644
--- a/drivers/net/wireless/ath/ath12k/dp_mon.c
+++ b/drivers/net/wireless/ath/ath12k/dp_mon.c
@@ -2325,7 +2325,6 @@ static int ath12k_dp_pkt_set_pktlen(struct sk_buff *skb, u32 len)
 			if ((pskb_expand_head(skb, 0,
 					      len - skb->len - skb_tailroom(skb),
 					      GFP_ATOMIC))) {
-				dev_kfree_skb_any(skb);
 				return -ENOMEM;
 			}
 		}
-- 
2.34.1

